File: HW02/plot.m
% plot_task.m

powers = 10:30;
times = [];

fid = fopen('task1.output', 'r');

for i = 1:length(powers)
    fgetl(fid);
    times(i) = str2double(fgetl(fid));
    fgetl(fid);
    fgetl(fid);
    fgetl(fid);
end

fclose(fid);

figure;
plot(powers, times, 'o');
xlabel('Size of the array (power)');
ylabel('Time (milliseconds)');
title('Scaling Analysis Plot');

saveas(gcf, 'task1.pdf');

File: HW02/plot.py
import matplotlib.pyplot as plt
from matplotlib.backends.backend_pdf import PdfPages

powers = range(10, 31)

times = []

with open("task1.txt") as f:
    for i in powers:
        f.readline()
        times.append(float(f.readline()))
        f.readline() 
        f.readline()  
        f.readline() 
        
        
with PdfPages("task1.pdf") as pdf:
    plt.plot(powers, times, "o-")
    plt.xlabel("Size of the array (power)")
    plt.ylabel("Time (ms)")
    plt.title("Scaling Analysis Plot")
    plt.xticks(powers)
    pdf.savefig()
    
